packages/mask/src/extension/popups/pages/Wallet/hooks/useWalletLockStatus.ts

```diff
@@ -1,21 +1,14 @@
-import { useAsyncRetry } from 'react-use'
 import { useEffect } from 'react'
 import Services from '#services'
 import { CrossIsolationMessages } from '@masknet/shared-base'
+import { useQuery } from '@tanstack/react-query'
 
 export function useWalletLockStatus() {
-    const {
-        value: isLocked,
-        loading,
-        error,
-        retry,
-    } = useAsyncRetry(async () => {
-        return Services.Wallet.isLocked()
-    }, [])
+    const { data: isLocked, isLoading: loading, error, refetch } = useQuery(['@@is-locked'], Services.Wallet.isLocked)
```

Comment:
why do you need to store the result of `isLocked`? it's a local query and it's fast. (and the value becomes invalid after refresh)

Reply:
It might be fast, but is still asynchronous with spinning loading consequently. But with persist cache, it becomes synchronous.

> the value becomes invalid after refresh

Would refetch to get latest status at startup.

It's an optimistic strategy to reuse previously cached status and recheck at startup.